sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1756,10 +1798,13 @@ public String apply(String input) {
 
       PCollection<TableRow> inputInGlobalWindow =
           input.apply(
-              Window.<TableRow>into(new GlobalWindows())
+              Window.<T>into(new GlobalWindows())
                   .triggering(DefaultTrigger.of())
-                  .discardingFiredPanes());
+                  .discardingFiredPanes())
+              .apply(MapElements.via(getFormatFunction())
```

Comment:
Can we skip over this `MapElements` if the result of `getFormatFunction()` equals `IDENTITY_FORMATTER`?

It's not unreasonable to not, as we get the appropriate behavior out, but we get to not include the extra step.